rotkehlchen/globaldb/handler.py

```diff
@@ -1036,8 +1037,23 @@ def edit_evm_token(entry: EvmToken) -> str:
                 f'due to a constraint being hit. Make sure the new values are valid ',
             ) from e
 
+        AssetResolver.clean_memory_cache(entry.identifier)
         return rotki_id
 
+    @staticmethod
+    def set_token_protocol_if_missing(token: EvmToken, new_protocol: str) -> None:
+        """Update the protocol of the evm token and clean the resolver cache"""
+        if token.protocol == new_protocol:
+            return
+
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            write_cursor.execute(
+                'UPDATE evm_tokens SET protocol = ? WHERE identifier = ?;',
+                (new_protocol, token.identifier),
+            )
+
+        AssetResolver.clean_memory_cache(token.identifier)
```

Comment:
setting these here both in this and edit function changes behaviour for many places. That's correct though right?